### DISCUSSION TITLE
What does header "X-ClientService-ClientTag" do?

### DISCUSSION THREAD

**Discussion by submarines-and** [pnp/pnpcore#679] [Q&A] [answered]
PnP adds header "x-clientservice-clienttag" when sending requests

https://github.com/pnp/pnpcore/blob/8c7d58762462cb6e62693c70ceaa98567edef134/src/sdk/PnP.Core/Services/Core/Http/SharePointRestClient.cs#L57

What does this header do? Does it have anything to do with bypassing throttling limits by decorating the traffic as described here?

https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#how-to-decorate-your-http-traffic-to-avoid-throttling

**Comment by jansenbe** [comment, marked as answer]
@submarines-and : it's for our internal telemetry needs: **every** call made to SharePoint Online is "logged" by our telemetry system, using this header we set a property on the request which allows us to filter out the requests made via PnP Core SDK. So this header has nothing to do with throttling handling.

**Reply by submarines-and** [reply]
Thank you for a quick response! :)
